File: Admin App/app/src/main/java/com/example/adminapp/models/ProjectModel.kt
package com.example.adminapp.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

@Entity(tableName = "Projects")
data class ProjectModel constructor(
    @PrimaryKey
    val id: UUID,
    val name: String,
    val description: String,
    val manager: String,
    val budget: Double,
    val status: Int,
    val startDate: Date,
    val endDate: Date,
    val specialRequirements: String?,
    val departmentInformation: String?
)

File: user_app/lib/models/project_model.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:user_app/converter/timestamp_converter.dart';
import 'package:user_app/converter/uuid_converter.dart';

part 'project_model.g.dart';

@JsonSerializable()
class ProjectModel {
  @UuidConverter() String? id;
  double? budget;
  String? departmentInformation;
  String? description;
  String? manager;
  String? name;
  String? othersList;
  String? specialRequirements;
  @TimestampConverter() DateTime? endDate;
  @TimestampConverter() DateTime? startDate;
  String? status;

  ProjectModel({
    this.id,
    this.budget,
    this.departmentInformation,
    this.description,
    this.manager,
    this.name,
    this.othersList,
    this.specialRequirements,
    this.endDate,
    this.startDate,
    this.status,
  });

  factory ProjectModel.fromJson(Map<String, dynamic> json) => _$ProjectModelFromJson(json);

  Map<String, dynamic> toJson() => _$ProjectModelToJson(this);
}
